### ISSUE TITLE
Using the React Rewrite of the DMN-Editor

### ISSUE THREAD

**Issue by aburdiak** [apache/incubator-kie-tools#2090]
This issue is related to the new React Rewrite of the DMN Editor https://github.com/apache/incubator-kie-issues/issues/439

- We have a need for a DMN Visualizer to be used in our React web app. We do not need editing capabilities
- We evaluated the `kie-editors-standalone` as the DMN editor fits our needs, but the package was too large to be used in our architecture (UI served from S3 bucket but proxied through a Lambda that sits behind an ALB - resulting in 1MB limit per file sent to user's browser -- [docs](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/lambda-functions.html#:~:text=The%20maximum%20size%20of%20the%20response%20JSON%20that%20the%20Lambda%20function%20can%20send%20is%201%20MB.))
- We learned of the React rewrite from this issue: apache/incubator-kie-issues#2034 and we plan to utilize this once it is complete
- In the meantime, after playing with the package, it looks to satisfy our needs in its current state. We are now looking into how we could use it. 

Main Questions
- What is needed to publish the existing state of the package to NPM to enable it to be pulled into a React UI? 
- Would it be possible to publish it in its current state as an Alpha build given it is not yet complete?
- Would you accept contributions to enable the package to be published in the near term?

@tiagobento tagging you as you look to be the right contact! Thanks!

**Comment by tiagobento** [comment]
Hi @aburdiak, thanks a lot for reaching out and also for evaluating the new DMN Editor. We're confident that we're making a lot of improvements compared to the current one, and it looks like you and your team already found these improvements to help in your use case. Glad to hear that, and thanks for sharing.

As for your questions:
> - What is needed to publish the existing state of the package to NPM to enable it to be pulled into a React UI?

Right now, we're finalizing the big PR that will introduce the new DMN Editor to the `main` branch. We expect that this PR will be merged this week. After that, we're gonna have to wait for the next release of Apache KIE to happen, so the packages will finally land on NPM. Unfortunately, we're not be able to provide an exact date yet, as there's a lot of conversations going on and also there's a lot of people working on making the release automations stable enough for a big release after we moved the KIE community to Apache. What I can tell you is that we expect this release will happen in January 2024.

> - Would it be possible to publish it in its current state as an Alpha build given it is not yet complete?

From our side, no. In the past we used to run KIE Tools as a separate stream compared to other KIE projects, but after the migration to Apache, we won't do that anymore. Apache KIE will have a single release stream, and KIE Tools is the "last piece of the puzzle". As of now, we don't have "dev"/"daily" releases, unfortunately, but it is definitely something we're going to look at in the future, as our users might want to be able to assess ongoing experimental modules such as the new DMN Editor.

> - Would you accept contributions to enable the package to be published in the near term?

At this moment, given the complexity of the entire ecosystem and also because we are also learning how we can achieve this first release in as an Apache project, I think contributing might be very hard. What I would suggest is that you stay in the loop. Asking questions, making suggestions, or even requesting information are all things that add a lot of value to the community. Sometimes we end up not documenting parts of the process and the operation of the projects that might be relevant to people depending on it, like you.


----

I'm sorry if those answers are not what you were expecting to hear, but I'm certain the Apache KIE community will gradually evolve to reaching a state where users can have a very good understanding of its release strategy as well as making it possible for people to participate in a more active way. Including experimental components/modules.

In the mean time, if you really can't wait for the next release to be out, my suggestion to you would be publishing your own build of the DMN Editor, with a different scope/name and a `pre-release` version discriminator. Like `@your-org/dmn-editor-experimental@0.0.0-dev1` or something along these lines :)

Of course, once we do release it to NPM you can drop this temporary workaround and start depending on the Apache KIE package.

**Comment by aburdiak** [comment]
@tiagobento Thank you so much for all the details! 

I assume this is the PR you are referring to right? apache/incubator-kie-issues#2064 

That's great to hear that you are hoping to have it published in January, we had assumed it would be towards the end of the first half of 2024 which would have been too long for us to wait. 

Your note about contributions makes a lot of sense and we are interested in being involved going forward. Once the package is published and we can utilize it in our project, we'll be sure to post any feedback.

I'll discuss with my team in regards to publishing it ourselves for the time being, but I would say waiting until the end of January would be the best overall decision.

**Comment by aburdiak** [comment]
@tiagobento I forgot to ask, is this the best repo to post issues to? Or should it be https://github.com/apache/incubator-kie-issues

**Comment by tiagobento** [comment]
@aburdiak 

Yep, that's the PR.

Glad to hear that! Contributions are always welcome!

Regarding issues, either repo is fine... we generally use https://github.com/apache/incubator-kie-issues but in reality we can aggregate everything at https://github.com/orgs/apache/projects/295 :)

**Comment by aburdiak** [comment]
@tiagobento 

Wanted to check in. Do you have an update regarding publishing the new editor under Apache?

**Comment by tiagobento** [comment]
@aburdiak Hi! We are still working on making the codebase ready to release under Apache.. you can follow the progress here: https://github.com/orgs/apache/projects/295/views/7?sliceBy%5Bvalue%5D=10.0.0. We don't have an official date yet, though..

**Comment by aburdiak** [comment]
@tiagobento Hi! 
Wanted to check in again. Would you have an estimate on when this might be published?

We are getting close to a point where we may end up publishing it ourselves in the interim but want to make sure we don't do that if not necessary.

I'm having a little trouble determining which task on the board relates to this release. Would tracking this 10.0.0 release be the right one? https://github.com/orgs/apache/projects/295/views/7?pane=issue&itemId=45202086&sliceBy%5Bvalue%5D=10.0.0

Thank you!!

**Comment by tiagobento** [comment]
@aburdiak The Apache KIE community needs to be release as a single unit, so there's other parts of the codebase holding it back. We're working hard to make everything ready for it, and the most important blocker we have right now is https://github.com/apache/incubator-kie-issues/issues/967. 

We're aiming at May 2024 for it now.... no guarantees, though :/

**Comment by aburdiak** [comment]
@tiagobento Thanks for the response! Totally understand the challenges of releasing it all and the coordination required to do that. 

We may be looking into publishing it ourselves temporarily. 

When I looked before, the way everything is packaged up was a little complicated. Would you be able to point me to any docs or portions of the repo where I could get more clarity on how its all packaged? Appreciate any help you can give, thank you!

**Comment by tiagobento** [comment]
@aburdiak How are you planning on using it? This will be important to know so I can help you... We didn't have time to document this stuff yet...

**Comment by aburdiak** [comment]
Hey @tiagobento We had some changes in our design that allowed us to continue without this package for the last few months. But we are getting to a point where we would ideally be able to use it soon. 

I know you all have done a ton of work on all of this and I'm excited to take advantage of it!

Would you have an updated estimate for when this package will be available? Thanks!

**Comment by tiagobento** [comment]
@aburdiak Hi! Apache KIE 10.0.0 is released. I came back here to let you know that you can now use the `@kie-tools/dmn-editor` package but unfortunately I discovered we had a problem during our publication. I reported

- https://github.com/apache/incubator-kie-issues/issues/1717

and we're currently investigating it. I'm confident we'll be able to publish this package soon and you'll be able to start using it after a long wait!

**Comment by tiagobento** [comment]
It took a while, but it's finally available with Apache KIE 10.1.0. See:
- https://www.npmjs.com/package/@kie-tools/dmn-editor-standalone

You can use it via jsDelivr too. See:
- https://github.com/apache/incubator-kie-tools/issues/3945#issue-3213707236
